api/controller/caasapplicationprovisioner/client.go

```diff
@@ -140,7 +139,8 @@ type ProvisioningInfo struct {
 	APIAddresses         []string
 	CACert               string
 	Tags                 map[string]string
-	Constraints          constraints.Value
+	Constraints          params.Value
+	CharmConstraints     params.CharmValue
```

Comment:
We don't want to leak params structs outside of the api layers. Use params structs over the wire and compose into a different bespoke struct to return to the caller. Sticking with the existing `constraints.Value` is what we want for `Constraints`. For `CharmConstraints` we can consider adding a new struct to `core/constraints` or `caas` since the values in this struct end up being assigned to a `caas.ApplicationConfig` struct

Reply:
reverted this per discussion offline